packages/build-utils/test/unit.lambda.test.ts

```diff
@@ -0,0 +1,94 @@
+import path from 'path';
+import { tmpdir } from 'os';
+import fs from 'fs-extra';
+import { createZip } from '../src/lambda';
+import { FileBlob, glob, spawnAsync } from '../src';
+
+describe('Lambda', () => {
+  it('should create zip file with symlinks', async () => {
+    if (process.platform === 'win32') {
+      console.log('Skipping test on windows');
+      return;
+    }
+    const files = await glob('**', path.join(__dirname, 'symlinks'));
+    expect(Object.keys(files)).toHaveLength(4);
+
+    const outFile = path.join(__dirname, 'symlinks.zip');
+    await fs.remove(outFile);
+
+    const outDir = path.join(__dirname, 'symlinks-out');
+    await fs.remove(outDir);
+    await fs.mkdirp(outDir);
+
+    await fs.writeFile(outFile, await createZip(files));
+    await spawnAsync('unzip', [outFile], { cwd: outDir });
+
+    const [linkStat, linkDirStat, aStat] = await Promise.all([
+      fs.lstat(path.join(outDir, 'link.txt')),
+      fs.lstat(path.join(outDir, 'link-dir')),
+      fs.lstat(path.join(outDir, 'a.txt')),
+    ]);
+    expect(linkStat.isSymbolicLink()).toEqual(true);
+    expect(linkDirStat.isSymbolicLink()).toEqual(true);
+    expect(aStat.isFile()).toEqual(true);
+  });
+
+  it('should create zip file with empty directory', async () => {
+    if (process.platform === 'win32') {
+      console.log('Skipping test on windows');
+      return;
+    }
+
+    const dir = await fs.mkdtemp(path.join(tmpdir(), 'create-zip-empty-dir'));
+    try {
+      const files = {
+        a: new FileBlob({
+          data: 'contents',
+          mode: 33188,
+        }),
+        empty: new FileBlob({
+          data: '',
+          mode: 16877,
+        }),
+        'b/a': new FileBlob({
+          data: 'inside dir b',
+          mode: 33188,
+        }),
+        c: new FileBlob({
+          data: '',
+          mode: 16877,
+        }),
+        'c/a': new FileBlob({
+          data: 'inside dir c',
+          mode: 33188,
```

Comment:
**suggestion (non-blocking):** label mode values

Can we label these mode values (with comments or named variables) to make it clear what each one means?